File: common_letter.py
teststr = "eaeeeeeedawdawdawd"


def common_letter(teststr):
    teststr = teststr.lower()

    letters = {}  # makes a dictionary of letters and their counts

    for letter in teststr:
        if letter in letters:
            letters[letter] += 1  # adds to count if letter already in dictionary
        else:
            letters[letter] = 1  # adds letter to dictionary if not already there

    maxletter = ""  # placeholder for most common letter name
    maxcount = 0  # placeholder for most common letter count
    for letter in letters:
        if (
            letters[letter] > maxcount
        ):  # checks if current letter is more common than previous most common letter
            maxletter = letter
            maxcount = letters[letter]  # sets new most common letter and count
    return maxletter


print(common_letter(teststr))

File: common_letter.js
let teststring = "aafbbccddaaaaadeeefff";

let common_letter = (str) => {
    lowerstr = str.toLowerCase();
    //find the most common letter in a string
    let lettercount = {};
    for (let i = 0; i < lowerstr.length; i++) {
        if (lettercount[lowerstr[i]]) {
            lettercount[lowerstr[i]]++;
        } else {
            lettercount[lowerstr[i]] = 1;
        }
    }
    let maxcount = 0;
    let maxletter = "";
    for (let letter in lettercount) {
        if (lettercount[letter] > maxcount) {
            maxcount = lettercount[letter];
            maxletter = letter;
        }
    }
    return maxletter + " appears " + maxcount + " times.";


}

console.log(common_letter(teststring));